scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/DatasetService.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    /**
+     * Finds all datasets in the database (Likely for admin use).
+     *
+     * @return Returns Page lists of all Datasets in the database.
+     */
+    @Override
+    public Page<Dataset> findAll() {
+        Pageable pageable = PageRequest.of(0, 20);
+        return datasetRepository.findAll(pageable);
+    }
+
+    /**
+     * Finds all Datasets in the database for a particular user
+     *
+     * @param user The user or owner of the Dataset.
+     * @return Returns Page lists of all Datasets in the database for a given user.
+     */
+    @Override
+    public Page<Dataset> findAllByUser(IUser user) {
+        Pageable pageable = PageRequest.of(0, 20);
+        return datasetRepository.findAllByUser(user, pageable);
+    }
+
+    /**
+     * Deletes a DataFile from a Dataset. This involves: 1. Removing the file from the appropriate
+     * location on the file system and 2. Removing the file from the database ensuring that the
+     * association with the parent Dataset is removed as well.
+     *
+     * @param datasetId The primary key of the Dataset
+     * @param fileId    The primary key of the DataFile
+     * @return boolean indicating success of deleting file
+     * @throws DatasetStorageException Exception detailing possible errors with the file and file system.
+     */
+    @Override
+    public boolean deleteFileFromDataset(Long datasetId, Long fileId) throws DatasetStorageException {
+
+        Optional<Dataset> dataset = datasetRepository.findById(datasetId);
+        Optional<DataFile> dataFile = dataFileRepository.findById(fileId);
+        if (dataset.isPresent() && dataFile.isPresent()) {
+            String username = dataset.get().getUsername();
+            String fileName = dataFile.get().getName();
+
+            // Remove join from dataset
+            dataset.get().getFiles().remove(dataFile.get());
+            datasetRepository.save(dataset.get());
+
+            // delete file from database
+            dataFileRepository.deleteById(fileId);
+
+            try {
+                // Removes file from file system
+                String pathString = fileSystemService.createPath(username,
+                        String.valueOf(datasetId)).toString();
+                File fileToBeDeleted = Paths.get(pathString, fileName).toFile();
+                return fileToBeDeleted.delete();
+            } catch (InvalidPathException e) {
+                throw new DatasetStorageException("Path to file could not be found", e);
+            } catch (UnsupportedOperationException e) {
+                throw new DatasetStorageException("Could not convert path to File", e);
+            } catch (SecurityException e) {
+                throw new DatasetStorageException("File was created, but could not be deleted", e);
+            }
+        } else {
+            throw new DatasetStorageException("Could not find DataFile/Dataset in the database");
+        }
```

Comment:
I would reverse the if logic 
```
if (!dataset.isPresent() || !dataFile.isPresent()) {
...
}
String...
```

It doesn't change what is happening, but it makes the code easier to read, since you don't have an else block.